packages/core/src/RPA/core/windows/locators.py

```diff
@@ -0,0 +1,358 @@
+import re
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+
+from RPA.core.locators import LocatorsDatabase, WindowsLocator
+
+from .helpers import IS_WINDOWS
+from .context import (
+    ElementNotFound,
+    WindowsContext,
+    WindowControlError,
+    with_timeout,
+)
+
+if IS_WINDOWS:
+    import uiautomation as auto
+    from uiautomation.uiautomation import Control
+
+
+WINDOWS_LOCATOR_STRATEGIES = {
+    "automationid": "AutomationId",
+    "id": "AutomationId",
+    "class": "ClassName",
+    "control": "ControlType",
+    "depth": "searchDepth",
+    "name": "Name",
+    "regex": "RegexName",
+    "subname": "SubName",
+    "type": "ControlType",
+    "index": "foundIndex",
+    "offset": "offset",
+    "desktop": "desktop",
+    "process": "process",
+    "handle": "handle",
+    "executable": "executable",
+}
+
+Locator = Union["WindowsElement", str]
+
+
+@dataclass
+class WindowsElement:
+    """Represent Control as dataclass"""
+
+    _WINDOW_SIBLING_COMPARE = {
+        "name": "name",  # this works for "subname" as well
+        "class": "class_name",
+        "control": "control_type",
+    }
+
+    item: Control
+    locator: Optional[Locator] = None
+    name: str = ""
+    automation_id: str = ""
+    control_type: str = ""
+    class_name: str = ""
+    left: int = -1
+    right: int = -1
+    top: int = -1
+    bottom: int = -1
+    width: int = -1
+    height: int = -1
+    xcenter: int = -1
+    ycenter: int = -1
+
+    def __init__(self, item, locator: Optional[Locator]):
+        self.item = item
+        self.locator: Optional[Locator] = locator
+        self.name = item.Name
+        self.automation_id = item.AutomationId
+        self.control_type = item.ControlTypeName
+        self.class_name = item.ClassName
+        rect = item.BoundingRectangle
+        self.left = rect.left if rect else -1
+        self.right = rect.right if rect else -1
+        self.top = rect.top if rect else -1
+        self.bottom = rect.bottom if rect else -1
+        self.width = rect.width() if rect else -1
+        self.height = rect.height() if rect else -1
+        self.xcenter = rect.xcenter() if rect else -1
+        self.ycenter = rect.ycenter() if rect else -1
```

Comment:
We could probably expose these directly instead of storing them like this. Maybe just have one `Region` object from core stored in the instance. No biggie though